File: hw/aimacana/u3/hw25CompositePattern/javascript/model/President.js
const Supervisor = require('./Supervisor');

class President extends Supervisor {
    static #instance = new President();
    
    constructor() {
        super();
        this.title = "President";
    }
    
    static getPresident(aName) {
        President.#instance.name = aName;
        return President.#instance;
    }
}
module.exports = President;

File: hw/aimacana/u3/hw25CompositePattern/python/model/president.py
from .supervisor import Supervisor

class President(Supervisor):
    _president = None
    
    def __new__(cls, *args, **kwargs):
        if not cls._president:
            cls._president = super(President, cls).__new__(cls)
        return cls._president

    def __init__(self, a_name=None):
        if not hasattr(self, 'initialized'):
            super().__init__()
            self.title = "President"
            self.initialized = True
            
    @classmethod
    def get_president(cls, a_name):
        inst = cls()
        inst.name = a_name
        return inst
